File: backend/app/schemas/integrity_alert.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class DatabaseIntegrityAlert(BaseModel):
    alert_type: str
    evidence_id: UUID | None
    evidence_number: str | None
    original_filename: str | None = None
    access_log_id: UUID | None = None
    access_session_ref: str | None = None
    detected_at: datetime
    database_hash: str | None = None
    blockchain_hash: str | None = None
    status: str


class DatabaseIntegrityAlertResponse(BaseModel):
    alert_count: int
    checked_count: int
    access_log_checked_count: int
    alerts: list[DatabaseIntegrityAlert]

File: frontend/src/interfaces/integrityAlert.ts
export interface DatabaseIntegrityAlert {
  alert_type: "EVIDENCE_HASH" | "ACCESS_LOG";
  evidence_id: string | null;
  evidence_number: string | null;
  original_filename: string | null;
  access_log_id: string | null;
  access_session_ref: string | null;
  detected_at: string;
  database_hash: string | null;
  blockchain_hash: string | null;
  status:
    | "DATABASE_HASH_MISMATCH"
    | "MISSING_ON_CHAIN"
    | "ACCESS_LOG_MISMATCH"
    | "ACCESS_LOG_MISSING_IN_DATABASE"
    | "ACCESS_LOG_MISSING_ON_CHAIN";
}

export interface DatabaseIntegrityAlertResponse {
  alert_count: number;
  checked_count: number;
  access_log_checked_count: number;
  alerts: DatabaseIntegrityAlert[];
}
